toolchain/parse/handle_expr.cpp

```diff
@@ -208,11 +208,16 @@ auto HandleExprInPostfix(Context& context) -> void {
       break;
     }
     default: {
+      CARBON_DIAGNOSTIC(ExpectedExpr, Error, "expected expression");
+      context.emitter().Emit(*context.position(), ExpectedExpr);
+
+      // Fallthrough to the error token case -- we don't need to diagnose those.
```

Comment:
Maybe "errors will have been diagnosed by lex" instead of "we don't need to diagnose those"?

```suggestion
// Fallthrough to the error token case -- errors will have been diagnosed by lex.
```